os_brick/initiator/connectors/lightos.py

```diff
@@ -187,66 +188,87 @@ def get_volume_paths(self, connection_properties):
+            if nvmec_match.match(match_path.split("/")[-2]):
+                continue
+
+            LOG.info("LIGHTOS: matching uuid %s was found"
+                     " for device path %s" % (uuid, match_path))
+            return os.path.join("/dev", match_path.split("/")[-2])
+        return None
 
+    @utils.trace
+    def _get_device_by_uuid(self, uuid):
+        endtime = time.time() + self.WAIT_DEVICE_TIMEOUT
+        while time.time() < endtime:
             try:
-                devname, err = self._execute('readlink', '-eq', lnk_path,
-                                             run_as_root=True,
-                                             root_helper=self._root_helper)
-                if devname.startswith("/dev/nvme"):
-                    devname = devname.strip()
-                    LOG.info("LIGHTOS: devpath %s detected for uuid %s",
-                             devname, uuid)
-                    return devname
+                device = self._check_device_exists_using_dev_lnk(uuid)
+                if device:
+                    return device
             except Exception as e:
-                LOG.debug(e)
-
-            for match_path in glob.glob(file_path):
-                try:
-                    match_wwid, err = self._execute(
-                        'cat',
-                        match_path,
-                        run_as_root=True,
-                        root_helper=self._root_helper)
-                except putils.ProcessExecutionError:
-                    LOG.warning("LIGHTOS: Could not find the uuid at %s",
-                                match_path)
-                    continue
-
-                if wwid != match_wwid.strip():
-                    continue
-
-                # skip slave nvme devices, for example: nvme0c0n1
-                if nvmec_match.match(match_path.split("/")[-2]):
-                    continue
-
-                LOG.info("LIGHTOS: matching uuid %s was found"
-                         " for device path %s" % (uuid, match_path))
-                return os.path.join("/dev", match_path.split("/")[-2])
+                LOG.debug(f'LIGHTOS: {e}')
+            device = self._check_device_exists_reading_block_class(uuid)
+            if device:
+                return device
 
             time.sleep(1)
         return None
 
+    def _verify_device_removed(self, uuid):
+        endtime = time.time() + self.WAIT_DEVICE_TIMEOUT
+        device_found_by_lnk = False
+        device_found_by_block_class = False
+        while time.time() < endtime:
+            try:
+                device_found_by_lnk = (
+                    self._check_device_exists_using_dev_lnk(uuid))
+            except Exception as e:
+                LOG.debug(f'LIGHTOS: {e}')
+            device_found_by_block_class = (
+                self._check_device_exists_reading_block_class(uuid))
+            if not device_found_by_block_class and not device_found_by_lnk:
```

Comment:
@yuval-lb why do we have/need exception handlign for _check_device_exists_using_dev_lnk
and not for _check_device_exists_reading_block_class